tests/unit/runtime/zero/test_zero_leaf_module.py

```diff
@@ -143,3 +188,79 @@ def test_set_no_match_class(self):
             raise AssertionError("Expected error that no module is set as a leaf module")
         except ValueError as e:
             pass
+
+
+class TestZ3LeafOptimization(DistributedTest):
+    world_size = 2
+    reuse_dist_env = True
+
+    def test_finegrained_optimization(self):
+        hidden_dim = 128
+        num_block = 16
+        stage3_module_granularity_threshold_list = [0, 100, 12100, 10000000]
```

Comment:
It is better to use `@pytest.mark.parametrize` for different threshold values. See following example: https://github.com/microsoft/DeepSpeed/blob/057d25be6775105f4b9e1d41e6c21981a157c849/tests/unit/runtime/zero/test_zero.py#L624

Reply:
thanks! modified.